itest/psbt_test.go

```diff
@@ -0,0 +1,180 @@
+package itest
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/lightninglabs/taro/internal/test"
+	"github.com/lightninglabs/taro/taropsbt"
+	"github.com/lightninglabs/taro/tarorpc"
+	wrpc "github.com/lightninglabs/taro/tarorpc/assetwalletrpc"
+	"github.com/lightninglabs/taro/taroscript"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/stretchr/testify/require"
+)
+
+// testPsbtSend tests that we can properly send assets back and forth between
+// nodes with the use of PSBTs.
+func testPsbtSend(t *harnessTest) {
+	// First, we'll make a normal asset with enough units to allow us to
+	// send it around a few times.
+	rpcAssets := mintAssetsConfirmBatch(
+		t, t.tarod, []*tarorpc.MintAssetRequest{simpleAssets[0]},
+	)
+
+	genInfo := rpcAssets[0].AssetGenesis
+	genBootstrap := genInfo.GenesisBootstrapInfo
+
+	ctxb := context.Background()
+
+	// Now that we have the asset created, we'll make a new node that'll
+	// serve as the node which'll receive the assets.
+	secondTarod := setupTarodHarness(
+		t.t, t, t.lndHarness.BackendCfg, t.lndHarness.Bob,
+		t.universeServer,
+	)
+	defer func() {
+		require.NoError(t.t, secondTarod.stop(true))
+	}()
+
+	var (
+		alice = t.tarod
+		// aliceLnd = t.lndHarness.Alice
+		bob    = secondTarod
+		bobLnd = t.lndHarness.Bob
+	)
+
+	// We need to derive two keys, one for the new script key and one for
+	// the internal key.
+	bobScriptKeyDesc, err := bobLnd.WalletKitClient.DeriveNextKey(
+		ctxb, &walletrpc.KeyReq{
+			KeyFamily: int32(taroscript.TaroKeyFamily),
+		},
+	)
+	require.NoError(t.t, err)
+	bobScriptInternalKey, err := btcec.ParsePubKey(
+		bobScriptKeyDesc.RawKeyBytes,
+	)
+	require.NoError(t.t, err)
+	bobInternalKeyDesc, err := bobLnd.WalletKitClient.DeriveNextKey(
```

Comment:
As in re-expose the lnd API used to create keys in the first place?

Reply:
Yeah, I think it could be useful to have this call be threaded through the daemon so we don't need to know what key family to use in `lnd` when talking to `tarod`. Will add that to the PR that adds the BTC level PSBT stuff (for funding and signing the anchor transactions externally).